sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -0,0 +1,282 @@
+        key, _, value = element.partition("=")
+        if key.lower() == "endpoint":
+            endpoint = value.rstrip("/")
+        elif key.lower() == "hostname":
+            endpoint = value.rstrip("/")
+        elif key.lower() == "sharedaccesskeyname":
+            shared_access_key_name = value
+        elif key.lower() == "sharedaccesskey":
+            shared_access_key = value
+        elif key.lower() == "entitypath":
+            entity_path = value
+    if not all([endpoint, shared_access_key_name, shared_access_key]):
+        raise ValueError(
+            "Invalid connection string. Should be in the format: "
+            "Endpoint=sb://<FQDN>/;SharedAccessKeyName=<KeyName>;SharedAccessKey=<KeyValue>"
+        )
+    entity = cast(str, entity_path)
+    left_slash_pos = cast(str, endpoint).find("//")
+    if left_slash_pos != -1:
+        host = cast(str, endpoint)[left_slash_pos + 2:]
+    else:
+        host = str(endpoint)
+    return host, str(shared_access_key_name), str(shared_access_key), entity
+
+
+def _generate_sas_token(uri, policy, key, expiry=None):
+    # type: (str, str, str, Optional[timedelta]) -> _AccessToken
+    """Create a shared access signiture token as a string literal.
+    :returns: SAS token as string literal.
+    :rtype: str
+    """
+    if not expiry:
+        expiry = timedelta(hours=1)  # Default to 1 hour.
+
+    abs_expiry = int(time.time()) + expiry.seconds
+    encoded_uri = quote_plus(uri).encode("utf-8")  # pylint: disable=no-member
+    encoded_policy = quote_plus(policy).encode("utf-8")  # pylint: disable=no-member
+    encoded_key = key.encode("utf-8")
+
+    token = utils.create_sas_token(encoded_policy, encoded_key, encoded_uri, expiry)
+    return _AccessToken(token=token, expires_on=abs_expiry)
+
+
+class ServiceBusSharedKeyCredential(object):
+    """The shared access key credential used for authentication.
+
+    :param str policy: The name of the shared access policy.
+    :param str key: The shared access key.
+    """
+
+    def __init__(self, policy, key):
+        # type: (str, str) -> None
+        self.policy = policy
+        self.key = key
+        self.token_type = b"servicebus.windows.net:sastoken"
+
+    def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
+        # type: (str, Any) -> _AccessToken
+        if not scopes:
+            raise ValueError("No token scope provided.")
+        return _generate_sas_token(scopes[0], self.policy, self.key)
+
+
+class BaseHandler(object):  # pylint:disable=too-many-instance-attributes
+    def __init__(
+        self,
+        fully_qualified_namespace,
+        entity_name,
+        credential,
+        **kwargs
+    ):
+        self.fully_qualified_namespace = fully_qualified_namespace
+        self._entity_name = entity_name
+        self._mgmt_target = self._entity_name + "/$management"
+        self._credential = credential
+        self._container_id = "servicebus.pysdk-" + str(uuid.uuid4())[:8]
```

Comment:
You have a constant for CONTAINER_PREFIX, but that points at eventhub.pysdk.  Should there be a constant for this/should that be servicebus?

Reply:
updated, thanks for calling out